storage-access-api/helpers.js

```diff
@@ -150,10 +150,32 @@ async function CanFrameWriteCookies(frame, keep_after_writing = false) {
   return can_write;
 }
 
+// Sets a cookie in an unpartitioned context by creating a new frame
+// and requesting storage access in the frame.
+async function SetUnpartitionedCookie(origin) {
+  let frame = await CreateFrame(`${origin}/storage-access-api/resources/script-with-cookie-header.py?script=embedded_responder.js`);
+  await SetPermissionInFrame(frame, [{ name: 'storage-access' }, 'granted']);
+  await RequestStorageAccessInFrame(frame);
+  await SetDocumentCookieFromFrame(frame, `cookie=unpartitioned;Secure;SameSite=None;Path=/`);
+  await SetPermissionInFrame(frame, [{ name: 'storage-access' }, 'prompt']);
+}
+
+// Tests for the presence of the unpartitioned cookie set by SetUnpartitionedCookie
+// in both the `document.cookie` variable and same-origin subresource \
+// Request Headers in the given frame
+async function HasUnpartitionedCookie(frame) {
+  let frameDocumentCookie = await GetJSCookiesFromFrame(frame);
+  let jsAccess = cookieStringHasCookie("cookie", "unpartitioned", frameDocumentCookie);
+  const httpCookie = await FetchSubresourceCookiesFromFrame(frame, "");
+  let httpAccess = cookieStringHasCookie("cookie", "unpartitioned", httpCookie);
+  assert_true(jsAccess == httpAccess, "HTTP and Javascript cookies must be in sync");
+  return jsAccess && httpAccess;
+}
+
 // Tests whether the current frame can read and write cookies via HTTP headers.
 // This deletes, writes, reads, then deletes a cookie named "cookie".
 async function CanAccessCookiesViaHTTP() {
-  await create_cookie(window.location.origin, "cookie", "1", "samesite=None;Secure");
+  await fetch(`${window.location.origin}/storage-access-api/resources/set-cookie-header.py?cookie=1;path=/;SameSite=None;Secure`);
```

Comment:
Can this just call `SetUnpartitionedCookie`, instead of duplicating the fetch?

Reply:
No, because we don't want to do the unpartitioning dance here. We want to test the current access status.

Reply:
Maybe add a comment here about that, to document the intent? E.g. `// We avoid reusing SetUnpartitionedCookie here, since that bypasses the cookie-accessibility settings that we want to check here.`.

Reply:
That sounds great.